### PR TITLE
Using reorder importers and yaml fmt pre-commit hook

### PR DESCRIPTION
**Pull request by unkcpz**

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -30,3 +30,8 @@ repos:
   rev: '3.9.2'
   hooks:
   - id: flake8
+
+- repo: https://github.com/asottile/reorder_python_imports
+  rev: v3.8.3
+  hooks:
+  - id: reorder-python-imports
```
Comment (CasperWA) [inline .pre-commit-config.yaml]: Please use `isort` instead, if you're using an import statement sorter.
```suggestion
  - repo: https://github.com/timothycrosley/isort
    rev: 5.10.1
    hooks:
    - id: isort
      args: ["--profile", "black", "--filter-files", "--skip-gitignore"]
```

**Review by CasperWA — CHANGES_REQUESTED** [review]

**Comment by CasperWA** [comment]
Is there a reason you're not using my code suggestion changes via GitHub and implementing the changes manually instead?

**Comment by unkcpz** [comment]
Hi @CasperWA, there is an indent in your suggestion. We don't use yaml lint repository. So I added it manually.

**Comment by unkcpz** [comment]
I can add the yaml lint here or a separate PR if you have no objection.

**Comment by CasperWA** [comment]
> I can add the yaml lint here or a separate PR if you have no objection.

Right - you could've just updated the suggested change to remove the indentation - but it's a recurring issue that is happening in other PRs as well, not just here it seems?

**Comment by unkcpz** [comment]
I add also the yaml formatter here.

Added: I also turn on the pre-commit ci for this repo.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -1,32 +1,43 @@
...
-  hooks:
-  - id: trailing-whitespace
-    exclude: README.md
-  - id: check-yaml
-  - id: check-json
-  - id: end-of-file-fixer
+    - repo: https://github.com/pre-commit/pre-commit-hooks
+      rev: v4.3.0
+      hooks:
+          - id: trailing-whitespace
```
Comment (CasperWA) [inline .pre-commit-config.yaml:6]:
```suggestion
          - id: trailing-whitespace
            args: [--markdown-linebreak-ext=md]
```

File: README.md [hunk 3]
```diff
@@ -21,7 +21,7 @@ Indeed, the filter grammar and parser and [`pydantic`](https://pydantic-docs.hel
 
 ## Prerequisites
 
-Environment where AiiDA is installed.  
+Environment where AiiDA is installed.
```
Comment (CasperWA) [inline README.md]: This trailing white-space should _not_ be removed (see the argument I have added to the respective hook - why did you remove the original exception there??)

**Review by CasperWA — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (unkcpz) [reply README.md]: Ah sorry, I simply didn't realize it is used here. Added back.

Comment (CasperWA) [reply README.md]: It's not just here - please add the extra argument I have put in my other comment above

Comment (unkcpz) [reply README.md]: That's true and using `--markdown-linebreak-ext=md` instead of excluding the `README.md` require double space to represent a newline. Changed.

Comment (CasperWA) [reply README.md]: There were double spaces there originally as well.

File: README.md [hunk 4]
```diff
@@ -160,7 +160,7 @@ To learn more about this, see the [`optimade-python-tools`](https://github.com/M
 
 ## For maintainers
 
-To release the new version, go to GitHub release API of the repo create a new release and update the release information. 
+To release the new version, go to GitHub release API of the repo create a new release and update the release information.
```
Comment (CasperWA) [inline README.md:163]: I think this is a case of where the trailing-whitespace hook correctly removed the trailing white space that wasn't meant to be there. I.e., I don't think a newline was intended here. But if you prefer it, please keep it, of course.
```suggestion
To release the new version, go to GitHub release API of the repo create a new release and update the release information.
```

**Comment by CasperWA** [comment]
There's no reason to add `pre-commit.ci` as well as running the `pre-commit` test job in the CI from the GitHub workflow "CI - Tests".

**Comment by unkcpz** [comment]
> There's no reason to add pre-commit.ci as well as running the pre-commit test job in the CI from the GitHub workflow "CI - Tests".

Thanks for pointing this out. Is it possible to remove `pre-commit` test job? It skip `codecov-validator`, is this able to be set for the `precommit.ci`?

**Comment by CasperWA** [comment]
> > There's no reason to add pre-commit.ci as well as running the pre-commit test job in the CI from the GitHub workflow "CI - Tests".
> 
> Thanks for pointing this out. Is it possible to remove `pre-commit` test job? It skip `codecov-validator`, is this able to be set for the `precommit.ci`?

I don't understand your second question.
If you meant that the codecov validation was skipped? Then this is a function of the `pytest (3.8)` test, which uploads the codecov report after a successful test run. If the pytest run does not succeed, no report will be uploaded.

To remove the `pre-commit` test job, simply remove it from the "CI - Tests" workflow.

Is there a particular reason you prefer using the `pre-commit.ci` job over the existing one?

**Comment by unkcpz** [comment]
> Is there a particular reason you prefer using the pre-commit.ci job over the existing one?

I simply found that `flake8` 3.9.2  is outdated and I think pre-commit.ci will automatically push to update it.
Moreover, the precommit.ci will auto fix the pre-commit error.

**Comment by CasperWA** [comment]
> > Is there a particular reason you prefer using the pre-commit.ci job over the existing one?
> 
> I simply found that `flake8` 3.9.2 is outdated and I think pre-commit.ci will automatically push to update it.

Updating the pre-commit hooks is implemented in the weekly dependency update workflow, see https://github.com/aiidateam/aiida-optimade/blob/develop/.github/workflows/ci_dependabot.yml#L57

If you wish to update them otherwise, just run `pre-commit autoupdate` locally and commit the (possible) changes to `.pre-commit-config.yaml` ...

**Comment by unkcpz** [comment]
Thanks! @CasperWA 
It seems the `pre-commit autoupdate` won't update the major version, but it makes sense. 

I think the weekly update bot is all good enough, I'll turn off precommit.ci back for the moment. The `precommi.ci` auto fix is not a big deal.

**Comment by CasperWA** [comment]
I was anyway thinking `flake8` should be changed out with `pylint`, but that's another task for another day.

**Comment by unkcpz** [comment]
> I was anyway thinking flake8 should be changed out with pylint, but that's another task for another day.

No problem. I turn off the precommit.ci, should be fine now.

**Review by CasperWA — APPROVED** [review]
I'm not sure I'm happy with so much indentation white space in the YAML files, but at least it's consistent now.
`isort` is something I've been meaning to implement for a long time as well, so cheers :)

**Comment by unkcpz** [comment]
@CasperWA thanks a lot for the checking and apologize for some careless fuzzy mistakes (back and forth) 😉
